circuits/types/field/src/from_bits.rs

```diff
@@ -31,7 +35,47 @@ impl<E: Environment> FromBits for Field<E> {
             // Check if all excess bits are zero.
             let should_be_zero = bits_le[size_in_bits..].iter().fold(Boolean::constant(false), |acc, bit| acc | bit);
             // Ensure `should_be_zero` is zero.
-            E::assert_eq(E::zero(), should_be_zero);
+            match should_be_zero.is_constant() {
+                true => match should_be_zero.eject_value() {
+                    false => (), // Constraint is satisfied.
+                    true => E::halt("Detected nonzero excess bits while initializing a base field element from bits."),
+                },
+                false => E::assert_eq(E::zero(), should_be_zero),
+            }
+        }
+
+        // If the number of bits is greater than `size_in_data_bits`, then check that it is a valid field element.
+        if num_bits > size_in_data_bits {
+            // Retrieve the modulus & subtract by 1 as we'll check `bits_le` is less than or *equal* to this value.
+            // (For advanced users) BaseField::MODULUS - 1 is equivalent to -1 in the field.
+            let modulus_minus_one = -E::BaseField::one();
+
+            // Pad `bits_le` with zeros to the size of the base field modulus, if necessary.
+            let boolean_false = Boolean::constant(false);
```

Comment:
It might be worth it to add a constant function `Boolean::false()` that returns `Boolean::constant(false)`

Reply:
Unfortunately, `false` is a reserved keyword in Rust

Reply:
we could try `fn True` and `fn False` (slightly unorthodox naming, but IMO clearer than `Boolean::constant(true)` and `Boolean::constant(false)`)

Reply:
Could we push this discussion to a new issue?